src/legal-officer/transaction-protection/LocContext.tsx

```diff
@@ -124,21 +138,105 @@ export interface Props {
+            fetchLocRequest(axios, contextValue.locId.toString())
+                .then(locRequest => {
+                    dispatch({ type: 'SET_LOC_REQUEST', locRequest });
+                })
+        }
+    }, [ contextValue.locRequest, contextValue.locId, axios ])
+
+    const enum NextRefresh {
+        STOP,
+        SCHEDULE,
+        IMMEDIATE
+    }
+
+    const refreshNameTimestamp = useCallback<() => Promise<NextRefresh>>(() => {
+        console.log(`refreshNameTimestamp() ${ refreshCounter } ${ new Date() }`)
+
+        function findItem(locRequest: LocRequest, item: LocItem): LocMetadataItem | LocFile | undefined {
+            if (item.type === 'Document') {
+                return findFile(locRequest, item.value)
+            } else {
+                return locRequest.metadata.find(metadata => metadata.name === item.name)
             }
         }
-    }, [ contextValue, openedLocRequests, closedLocRequests ])
+
+        function findFile(locRequest: LocRequest, hash: string): LocFile | undefined {
+            return locRequest.files.find(file => file.hash === hash)
+        }
+
+        function allItemsOK(items: LocItem[]): boolean {
+            return items.find(item => item.status === "PUBLISHED" && (item.name === UNKNOWN_FILE || item.timestamp === null)) === undefined
+        }
+
+        if (contextValue.loc === null || axios === undefined) {
+            return Promise.resolve(NextRefresh.SCHEDULE);
+        }
+
+        if (allItemsOK(contextValue.locItems)) {
+            return Promise.resolve(NextRefresh.STOP);
+        }
+
+        const proceed = async () => {
+            let nextRefresh = NextRefresh.SCHEDULE;
+            const locRequest = await fetchLocRequest(axios, contextValue.locId.toString());
+            contextValue.locItems
+                .filter(locItem => locItem.type === 'Document' && locItem.name === UNKNOWN_FILE)
+                .forEach(locItem => {
+                    const locFile = findFile(locRequest, locItem.value)
+                    if (locFile && locFile.name) {
+                        nextRefresh = NextRefresh.IMMEDIATE;
+                        dispatch({ type: 'UPDATE_ITEM_NAME', locItem, name: locFile.name })
+                    }
+                })
+            contextValue.locItems
+                .filter(locItem => locItem.timestamp === null)
+                .forEach(locItem => {
+                    const locFile = findItem(locRequest, locItem)
+                    if (locFile && locFile.addedOn) {
+                        nextRefresh = NextRefresh.IMMEDIATE;
+                        dispatch({ type: 'UPDATE_ITEM_TIMESTAMP', locItem, timestamp: locFile.addedOn })
+                    }
+                })
+            return nextRefresh;
+        }
+        return proceed()
+    }, [ contextValue.loc, contextValue.locItems, contextValue.locId, axios, refreshCounter, NextRefresh.STOP, NextRefresh.SCHEDULE, NextRefresh.IMMEDIATE])
```

Comment:
I think that moving the enum definition out of the component will enable you to remove the enum items from the dependency array.